### DISCUSSION TITLE
Make volume slider remappable

### DISCUSSION THREAD

**Discussion by NewsGuyTor** [music-assistant/support#797] [Feature requests and ideas]
I really hope you can add an option to remap the volume slider, how it works now is damn scary. 

On my Samsung HW-Q90R soundbar volume 10 is okay for chilling, volume 13 for normal listening, at volume 20 I'm worried about my neighbors and at volume 27 I'm worried about my own hearing.

There needs to be a way to remap it, so I can set volume 24 for example to be 100. How it works now is downright scary, it's way too sensitive and can play way too loudly.

**Comment by marcelveldt** [comment]
Can you please rephrase this a bit ?

**Reply by NewsGuyTor** [reply]
Volume 20 on my speaker is window shaking LOUD, but the volume slider on Music Assistant seem to think that volume 20 is 20% of the volume. 

There should be an option to configure it so that, for example, volume 20 is 100%, so when you click in the middle of the volume slider you set the volume to 10.

**Reply by marcelveldt** [reply]
Can you please check the HA integration for this speaker/player if that has the same issue ?
This sounds like the wrong mapping is already there ? Better fix it there than in MA.

**Reply by NewsGuyTor** [reply]
I'm a bit confused about what you mean, the soundbar is a device owned by the Music Assistant integration.
<img width="916" alt="Skjermbilde 2022-08-04 kl  08 41 49" src="https://user-images.githubusercontent.com/19603024/182780609-d80b0a3f-35b2-4fa9-9163-fcb3c574c0a4.png">

**Reply by marcelveldt** [reply]
No, Music Assistant is based on Home Assistant and gets the devices from Home Assistant. So what I'm saying is that perhaps you should have a look at the original Home Assistant entity for this soundbar if this has the same issue.

**Reply by marcelveldt** [reply]
That screenshot is from the "resulting" device created by Music Assistant. You also have a source device somewhere. Maybe DLNA ?

**Reply by NewsGuyTor** [reply]
Ah, yeah, I have a DLNA device too. 

That said, I don't think it can be fixed there, I've gotten the impression that DLNA is pretty basic and dumb.

The mapping isn't technically wrong after all, it's just scary and naive. With MA a better and less naive volume slider could be implemented, I think.

**Reply by marcelveldt** [reply]
Maybe it can be as simple as a player option to set the maximum volume and the volume slider will not only account for that (and treat that value as 100%) but also the rest of the logic will make sure no higher values are sent. In that case it also acts as a safe guard

**Reply by NewsGuyTor** [reply]
That sounds great!

**Comment by OzGav** [comment]
Relates to v1
